gluoncv/auto/data/dataset.py

```diff
@@ -385,6 +397,42 @@ def __getitem__(self, idx):
             label = self._dataset['label'][idx]
         return img, label
 
+class _TorchImageClassificationDataset(TorchDataset):
+    """Internal wrapper read entries in pd.DataFrame as images/labels.
+
+    Parameters
+    ----------
+    dataset : ImageClassificationDataset
+        DataFrame as ImageClassificationDataset.
+
+    """
+    def __init__(self, dataset):
+        if torch is None:
+            raise RuntimeError('Unable to import pytorch which is required.')
+        assert isinstance(dataset, ImageClassificationDataset)
+        assert 'image' in dataset.columns
+        self._has_label = 'label' in dataset.columns
+        self._dataset = dataset
+        self.classes = self._dataset.classes
+        self._imread = Image.open
+        self.transform = None
+
+    def __len__(self):
+        return self._dataset.shape[0]
+
+    def __getitem__(self, idx):
+        im_path = self._dataset['image'][idx]
+        img = self._imread(im_path).convert('RGB')
+        label = None
+        # # pylint: disable=not-callable
+        if self.transform is not None:
+            img = self.transform(img)
+        if self._has_label:
+            label = self._dataset['label'][idx]
+        else:
+            label = torch.tensor(-1, dtype=torch.long)
```

Comment:
Is this -1 label used anywhere? What will timm do in such case?

Reply:
I'm just following timm's dataset implementation. The data loader requires label to be a tensor instead of None, and timm's dataset will give a -1 tensor in such case.